src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectDependenciesPackageCommandResolver.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal class ProjectDependenciesPackageCommandResolver : ICommandResolver
+    {
+        private List<string> _allowedCommandExtensions;
+        private IPackagedCommandSpecCreator _packagedCommandSpecCreator;
+
+        public ProjectDependenciesPackageCommandResolver(IPackagedCommandSpecCreator packagedCommandSpecCreator)
+        {
+            _packagedCommandSpecCreator = packagedCommandSpecCreator;
+        }
+
+        public CommandSpec Resolve(CommandResolverArguments commandResolverArguments)
+        {
```

Comment:
This method seems pretty big. Try to break it apart in smaller methods and just call them here.